x-pack/platform/packages/shared/kbn-streamlang/src/transpilers/ingest_pipeline/conversions.ts

```diff
@@ -66,6 +70,12 @@ export function convertStreamlangDSLActionsToIngestPipelineProcessors(
       );
     }
 
+    if (action === 'drop_document') {
```

Comment:
For `replace` processor, I've too encountered the situation where the Streamlang processor's name is different than the underlying Ingest Processor. There, I've introduced [`processorRenames`](https://github.com/elastic/kibana/pull/242310/files#diff-9ffe177fc26d8fc31facc38295392b57b3712c68e059ef5c132fc32796d876a5R46) to tackle that in the pre processing step, like field renames are handled. See if it makes sense to follow that approach for `drop_document` too.